src/app/shared/components/file-preview/file-preview.component.html

```diff
@@ -0,0 +1,31 @@
+<div [ngSwitch]="mediaType">
+    <div *ngSwitchCase="'image'">
+        <img class="preview"
+            [src]="url"
+            (load)="loadingPreview = false"
+            alt="" />
+    </div>
+    <div *ngSwitchCase="'video'">
+        <video class="preview"
+               width="100%"
+               controls
+               preload="metadata"
+               controlsList="nodownload"
+               (loadedmetadata)="loadingPreview = false">
+            <source [src]="url">
+        </video>
+    </div>
+
+    <div *ngSwitchCase="'audio'" class="preview audio">
+        <mesh-audio-play-button [src]="url"
+                                [autoPlay]="false"
+                                (load)="loadingPreview = false"></mesh-audio-play-button>
+    </div>
+
+    <div *ngSwitchDefault>
+        <div class="default-preview preview">
+            <icon class="node-icon">insert_drive_file</icon>
```

Comment:
The existing file placeholder I made for the BinaryFieldComponent also included the mime type overlaid over the file icon and a bunch of styling which has not made it over to this component.